tests/Blueprints/Runner/Step/UnzipStepRunnerTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Blueprints\Runner\Step;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Filesystem\Path;
+use WordPress\Blueprints\Model\DataClass\UnzipStep;
+use WordPress\Blueprints\Progress\Tracker;
+use WordPress\Blueprints\Resources\ResourceManager;
+use WordPress\Blueprints\Runner\Step\UnzipStepRunner;
+use PHPUnit\Framework\TestCase;
+use WordPress\Blueprints\Runtime\Runtime;
+
+class UnzipStepRunnerTest extends TestCase {
+
+
+	/**
+	 * @var string
+	 */
+	private string $document_root;
+
+	/**
+	 * @var Runtime
+	 */
+	private Runtime $runtime;
+
+	/**
+	 * @var UnzipStepRunner
+	 */
+	private UnzipStepRunner $step;
+
+	/**
+	 * @var Filesystem
+	 */
+	private Filesystem $file_system;
+
+	/**
+	 * @before
+	 */
+	public function before() {
+		$this->document_root = Path::makeAbsolute( 'test', sys_get_temp_dir() );
+		$this->runtime       = new Runtime( $this->document_root );
+
+		$resource_manager = $this->createStub( ResourceManager::class );
+		$resource_manager->method( 'getStream' )
+			->willReturn( fopen( __DIR__ . '/test.zip', 'rb' ) );
+
+		$this->step = new UnzipStepRunner();
+		$this->step->setRuntime( $this->runtime );
+		$this->step->setResourceManager( $resource_manager );
+
+		$this->file_system = new Filesystem();
+	}
+
+	/**
+	 * @after
+	 */
+	public function after() {
+		$this->file_system->remove( $this->document_root );
+	}
+
+	public function testUnzipFileWhenUsingAbsolutePath(): void {
+		$input = new UnzipStep();
+		$zip   = __DIR__ . '/test.zip';
+		$input->setZipFile( $zip );
+		$relative_path = 'dir';
+		$absolute_path = $this->runtime->resolvePath( $relative_path );
+		$input->setExtractToPath( $absolute_path );
+
+		$this->step->run( $input, new Tracker() );
+
+		$this->assertDirectoryExists( $absolute_path );
```

Comment:
Let's check if the directory actually contains the expected files and assert their content is as expected.

Reply:
Done.